src/game/level_update.c

```diff
@@ -608,7 +606,7 @@ s16 music_unchanged_through_warp(s16 arg) {
 void initiate_warp(s16 destLevel, s16 destArea, s16 destWarpNode, s32 warpFlags) {
     if (destWarpNode >= WARP_NODE_CREDITS_MIN) {
         sWarpDest.type = WARP_TYPE_CHANGE_LEVEL;
-    } else if (destLevel == EXIT_COURSE_LEVEL && destArea == EXIT_COURSE_AREA && destWarpNode == EXIT_COURSE_NODE) {
+    } else if (warpFlags == WARP_FLAG_EXIT_COURSE) {
```

Comment:
Non-blocking, but I'm wondering if we should conjoin these conditions into one to reduce the number of lines here. I could see that going either way for readability but worth mentioning at least.

Reply:
the precedent has it split up for readability so i just went with that as well